subprojects/code-quality/src/main/groovy/org/gradle/api/plugins/quality/PmdPlugin.java

```diff
@@ -146,7 +155,15 @@ protected void configureForSourceSet(final SourceSet sourceSet, final Pmd task)
         task.setDescription("Run PMD analysis for " + sourceSet.getName() + " classes");
         task.setSource(sourceSet.getAllJava());
         ConventionMapping taskMapping = task.getConventionMapping();
-        taskMapping.map("classpath", () ->
-            sourceSet.getOutput().plus(sourceSet.getCompileClasspath()));
+        ConfigurationContainer configurations = project.getConfigurations();
+        Configuration compileConfiguration = configurations.getByName(sourceSet.getCompileClasspathConfigurationName());
+        Configuration runtimeConfiguration = configurations.getByName(sourceSet.getRuntimeClasspathConfigurationName());
+        Configuration pmdAuxClasspath = configurations.create(sourceSet.getName() + "PmdAuxClasspath");
```

Comment:
I would consider also adding a bucket configuration which can be used to declare additional dependencies required for PMD, like `compileOnly` dependencies of transitives.

Reply:
You mean, expose this configuration (or some other that this one would extend) as part of the plugin API, alongside the existing `pmd`?

Reply:
I mean documenting how to add more entries to the auxiliary classpath of PMD. And ideally we should use a bucket configuration for this, not directly the resolvable one.
And it would be documented [next to `pmd`](https://docs.gradle.org/current/userguide/pmd_plugin.html#sec:pmd_dependency_management).
Makes sense?

Reply:
I've added `pmdAux` configuration for this.